File: Problem1.java
/*
Broken Calculator
approach: we can use bfs but that would be exponential, we can optimize by going from target to start value
time: log target
space: O(1)
 */
public class Problem1 {
    public static int brokenCalc(int startValue, int target) {
        int op = 0;
        while (target > startValue) {
            if (target%2 == 0) {
                target/=2;
            }
            else {
                target+=1;
            }
            op++;
        }
        return op+startValue-target;
    }

    public static void main(String[] args) {
        System.out.println(brokenCalc(3,10));
    }
}

File: Problem_1.js
// ## Problem1: Calculate Using Broken Calculator (https://leetcode.com/problems/broken-calculator/)

// TC: O(log(target))
// SC: O(1)

// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

// Your code here along with comments explaining your approach

/**
 * @param {number} startValue
 * @param {number} target
 * @return {number}
 */
var brokenCalc = function (startValue, target) {
    let operations = 0;

    while (target > startValue) {
        if (target % 2 === 0) {
            // If target is even, divide by 2
            target = target / 2;
        } else {
            // Else add 1
            target = target + 1;
        }
        operations++;
    }
    // Only operations possible are adding
    return operations + (startValue - target);
};